rfc/cdevents-spinnaker.md

```diff
@@ -0,0 +1,106 @@
+Spinnaker can produce and consume [CDEvents](https://cdevents.dev/) using [Java SDK](https://github.com/cdevents/sdk-java) which is developed and maintained by the [CDEvents community](https://cdevents.dev/community/).
+
+### Goals and Non-Goals
+Goals:
+- Event-Driven Standardization with CI/CD tools like Keptn, Tekton, Jenkins, Spinnaker, etc..
+- Create an `events-broker` that is able to produce and consume `CDEvents`
+- One needs to subscribe to `events-broker` with the type of event that needs to consume
+- API end points will be created to produce and consume `CDEvents`
+
+## Motivation and Rationale
+
+- Currently there is no common specification for CI/CD tools to communicate with in the CI/CD ecosystem. 
+
+- In a complex and fast-moving CI/CD world with a lot of different tools and platforms that need to communicate with each other.
+
+- Multiple pipelines can be run between CI/CD tools by sharing the artifacts and its metadata.
+
+- Increase in traceability with in the ecosystem from commit to deployment of an artifact.
+
+- Operators can deploy Spinnaker with `CDEvents` to run multiple pipelines using different CI/CD tools in ecosystem.
+
+## Timeline
+Proof of concept with Spinnaker, Keptn and Tekton CI/CD tools developed to communicate with each other about occurrences and running pipelines from building an artifact to deploy.
+
+The final implementation aligning with `CDEvents` vocabulary specification will take from 4-6 weeks.
+
+## Design
+Proposing a design based on the PoC outcome.
+
+A `CDEventsController` will be created with the API endpoints to produce and consume `CDEvents` from `events-broker` with request mappings,
+
+`@RequestMapping(value = "cdevents/consume", method = RequestMethod.POST)`
+
+`@RequestMapping(value = "cdevents/produce", method = RequestMethod.POST)`
+
+
+An `events-broker` will be created to send and receive events by knative-eventing and the sample URL can be configured as below in the cluster.
+
+`events-broker : http://broker-ingress.knative-eventing.svc.cluster.local/default/events-broker`
+
+#### Consume CDEvents
+
+Spinnaker API endpoint `cdevents/consume` will be subscribed to `events-broker` to listen to the various types of events and Spinnaker Pipelines need to be configured with the `CDEvents` type on which Spinnaker is interested on.
+Example: A sample event `dev.cdevents.artifact.published` subscribed to `events-broker`
+``` 
+kubectl create -f - <<EOF
+apiVersion: eventing.knative.dev/v1
+kind: Trigger
+metadata:
+  name: cd-artifact-published-to-spinnaker
+spec:
+  broker: events-broker
+  filter:
+    attributes:
+      type: dev.cdevents.artifact.published
+  subscriber:
+    uri: http://<spin-service>:<port>/cdevents/consume
+EOF 
+```
+Once the event is received by Spinnaker API endpoint `cdevents/consume`, further can be handled to run pipelines based on the configured event data in Spinnaker.
```

Comment:
Is the intention that the (and making an assumption that it'd be gate -> echo) endpoint would return in YAML?  Is this API defined/accepted?  What's the maintenance path on these APIs?  E.g. how often do these API's change, are there SDK's needing updated, dependency impacts?